common-types/src/main/scala/com/improving/app/common/errors/Validation.scala

```diff
@@ -1,85 +1,97 @@
+  val urlValidator: Validator[String] = url => {
+    None // TODO
+  }
 
-    def skipEmpty(validator: Validation.Validator[String]): Validator[String] = str => {
-      if(str.isEmpty) {
-        None
-      } else {
-        validator(str)
-      }
+  val tenantIdValidator: Validator[TenantId] = tenantId => {
+    if (tenantId.id.isEmpty) {
+      Some(ValidationError("Tenant Id is empty"))
+    } else {
+      None
     }
+  }
 
-    val tenantIdValidator: Validator[TenantId] = tenantId => {
-      if (tenantId.id.isEmpty) {
-        Some(ValidationError("Tenant Id is empty"))
-      } else {
-        None
-      }
+  val memberIdValidator: Validator[MemberId] = memberId => {
+    if (memberId.id.isEmpty) {
+      Some(ValidationError("Member Id is empty"))
+    } else {
+      None
     }
+  }
 
-    val memberIdValidator: Validator[MemberId] = memberId => {
-      if (memberId.id.isEmpty) {
-        Some(ValidationError("Member Id is empty"))
-      } else {
-        None
-      }
+  val organizationIdValidator: Validator[OrganizationId] = organizationId => {
+    if (organizationId.id.isEmpty) {
+      Some(ValidationError("Organization Id is empty"))
+    } else {
+      None
     }
+  }
 
-    val contactValidator: Validator[Contact] = contact => {
-      if (
-        contact.firstName.isEmpty ||
-          contact.lastName.isEmpty ||
-          contact.emailAddress.forall(_.isEmpty) ||
-          contact.phone.forall(_.isEmpty) ||
-          contact.userName.isEmpty
-      ) {
-        Some(ValidationError("Primary contact info is not complete"))
-      } else {
-        None
-      }
+  val contactValidator: Validator[Contact] = contact => {
+    if (
+      contact.firstName.isEmpty ||
+        contact.lastName.isEmpty ||
+        contact.emailAddress.forall(_.isEmpty) ||
+        contact.phone.forall(_.isEmpty) ||
+        contact.userName.isEmpty
+    ) {
+      Some(ValidationError("Primary contact info is not complete"))
+    } else {
+      None
     }
+  }
 
-    val addressValidator: Validator[Address] = address => {
-      val isPostalCodeMissing = address.postalCode.fold(true) {
-        postalCode =>
-          postalCode.postalCodeValue match {
-            case UsPostalCodeImpl(code) => code.isEmpty
-            case CaPostalCodeImpl(code) => code.isEmpty
-          }
-      }
-      if (
-        address.line1.isEmpty ||
-          address.city.isEmpty ||
-          address.stateProvince.isEmpty ||
-          address.country.isEmpty ||
-          isPostalCodeMissing
-      ) {
-        Some(ValidationError("Address information is not complete"))
-      } else {
-        None
-      }
+  val addressValidator: Validator[Address] = address => {
+    val isPostalCodeMissing = address.postalCode.fold(true) {
+      postalCode =>
+        postalCode.postalCodeValue match {
+          case UsPostalCodeImpl(code) => code.isEmpty
```

Comment:
Would be nice if we were also validating patterns eg US is 5 digits. Can be done in another story if desired

Reply:
I created IA-168 to flesh out shared validation code